crates/ruff/src/rules/pep8_naming/rules/dunder_function_name.rs

```diff
@@ -0,0 +1,72 @@
+use ruff_macros::{define_violation, derive_message_formats};
+use rustpython_parser::ast::Stmt;
+
+use crate::ast::helpers::identifier_range;
+use crate::ast::types::{Scope, ScopeKind};
+use crate::registry::Diagnostic;
+use crate::source_code::Locator;
+use crate::violation::Violation;
+
+define_violation!(
+    /// ## What it does
+    ///
+    /// Checks for functions with “dunder” names (i.e. names with two leading and two trailing underscores).
```

Comment:
This sounds like it also detects well-defined magic methods such as `__len__`.

Reply:
It checks only for functions, not for methods. Afaik only `__getattr__ ` and `__dir__` are allowed (https://peps.python.org/pep-0562/)